src/ztex_md5crypt.c

```diff
@@ -131,16 +131,22 @@ static void init(struct fmt_main *fmt_main)
 static int crypt_all(int *pcount, struct db_salt *salt)
 {
 	int result;
-	unsigned char salt_buf[18]; // salt to send to device
+	unsigned char salt_buf[22]; // salt to send to device
 
 	int salt_len = strnlen(salt->salt, 8);
 
 	// 1 byte unused, 1 byte salt_len, 16 bytes salt in network byte order
 	salt_buf[0] = 0;
 	salt_buf[1] = salt_len;
 	memcpy(salt_buf + 2, salt->salt, salt_len);
+	// The device expects salt (18 bytes) then 4 bytes for each
+	// "tunable cost", for those found in struct fmt_main.
+	// We add-up "tunable cost" of 1,000 like if it was in "struct fmt_main".
+	salt_buf[18] = 1000 % 256;
+	salt_buf[19] = 1000 / 256;
+	*((uint16_t *)(salt_buf + 20)) = 0;
```

Comment:
This is a bug: `salt_buf` is not guaranteed to be aligned to allow for the 16-bit write. Please use two single-byte writes instead.